### ISSUE TITLE
Postman Collection Level Pre Scripts request runs fine in Postman but failing in Newman 

### ISSUE THREAD

**Issue by mianamirrashid** [postmanlabs/newman#1713]
1. Newman Version (can be found via `newman -v`): 4.1.0
 2. OS details (type, version, and architecture): Windows 8
 3. Are you using Newman as a library, or via the CLI? CLI
 3. Did you encounter this recently, or has this bug always been there:Just start using it in CLI 
 4. Expected behaviour: For "Pre Request - Script"  should be able to get resposne body and save that value in Environment variables for later usage.  
 5. Command / script used to run Newman: 
 6. Sample collection, and auxiliary files (minus the sensitive details):
 7. Screenshots (if applicable):

**Comment by codenirvana** [comment]
@mianamirrashid Can you share a sample collection (minus the sensitive details) with this behavior?

**Comment by mianamirrashid** [comment]
Folder A (Parent ) - Pre-request Scripts (Nothing else in it ) 

const echoRequestfortoken  = {
   
    url : "https://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    method : 'POST',
    header: {Accept:"application/json","Content-Type":"application/json",   "Api_key": "xxxxxxxxxxxxxxxx", "cache-control":"no-cache"    },
    body: {
        //mode:'raw',
        mode: 'application/json',
        raw: JSON.stringify(
            { customerId:pm.globals.get("CustomerId"), role:"fullaccessaccount-01"    
        })
    }
};

pm.sendRequest ( echoRequestfortoken , function (err, res){
    var jsonResponse = (res.text());
    var getvalue = eval ('(' + jsonResponse + ')');
    var response_body_only = getvalue.accessToken;
    pm.environment.set("AccessToken", getvalue.accessToken);
   
});


Folder B (Child) Pre-request Scripts  
const echoRequestforhandshake  = {
   url : "https://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    method : 'POST',
    header: {Accept:"application/json","Content-Type":"application/json",   "Api_key": "xxxxxxxxxxxxxxxxxxx", "Authorization": "Bearer " + pm.environment.get("AccessToken"),"cache-control":"no-cache"     },
    body: {
        //mode:'raw',
        mode: 'application/json',
        raw: JSON.stringify(
            { customerId:pm.globals.get("CustomerId"),    Password: pm.globals.get("Password"),           accessType:"PPRO", accessmode:"Internet"     
        })
    }
};
    
    pm.sendRequest ( echoRequestforhandshake , function (err, res){
    var jsonResponse = (res.text());
    var getvalue = eval ('(' + jsonResponse + ')');
    var response_body_only = getvalue.accountSessionId;
    pm.environment.set("SessionId", getvalue.accountSessionId);
    
});


Inside this folder some requests nothing special 

Same scripts works fine in Postman but in Newman pre-Script for forlder not able to hold response body.

**Comment by Manojkumar05** [comment]
Newman Version (can be found via newman -v): 4.1.0
OS details (type, version, and architecture): Windows 7
Are you using Newman as a library, or via the CLI? Library in Node Modules
Did you encounter this recently, or has this bug always been there:Just started using 
Expected behaviour: **As it is working fine in postman, The same is expected in Newman.**
Sample collection, and auxiliary files (minus the sensitive details):Unable to attach from my environment

My input data as .csv file - When I previewed file  from postman collection
" {\n \"FolderId\" : 102,\n \"FolderName\" : \"Manoj_root\"\n }"

**### IN POSTMAN**
1)In Pre Request script
var AddFolderTo_req=JSON.parse(JSON.stringify(pm.iterationData.get("AddFolderTo")));
pm.environment.set("AddFolderTo_req",AddFolderTo_req);

> Above parse and stringify are using to remove new lines and then to stringify

2)In Body
{{AddFolderTo_req}}

3)In Test
pm.test("Status code is 200: Add folder to root", function () {
    pm.response.to.have.status(200);
});

I Exported Environment file, Input data file and Collection file and started using in Newman
Below is the error encountered

`UnhandledPromiseRejectionWarning: Error : Parse Error: expected:  '"' got: ':' . at ': {\n\`
at parseEscapedItem: fast-csv\lib\parser\parser.js:80:19
at parseStream.parseLine: parse.js: 192:30
at parser_stream.js

**Comment by Manojkumar05** [comment]
Please find attached zip files for your reference

[AddFolder.zip](https://github.com/postmanlabs/newman/files/2382857/AddFolder.zip)

**Comment by codenirvana** [comment]
@Manojkumar05 This works fine for me, can you share the script used to run Newman?

**Comment by Manojkumar05** [comment]
`newman.run({
            collection: require(collectionPath),
            iterationData: "D:\\Work\\Automation\\API\\testdata.csv",
            environment : "D:\\Work\\Automation\\API\\Variables\\EnvVariables\\Env1.json" ,
            iterationCount: iterCnt,
            asLibrary: true,
            stopOnError: false,
            reporters: 'html',
            reporter: { "html": { "export": exportResultPath + ".html" } }
        }, function (err, summary) {
            if (err) { throw err; }
            console.log('collection run complete');

        })`

1)iterationData - Input csv data which contains Input JSON.
2)environment - Exported environement variables JSON from postman 
3)collectionPath - which is path of collecttion

**Comment by Manojkumar05** [comment]
I am facing parsing error issue for both input data in csv or json format.

Error Information:
UnhandledPromiseRejectionWarning: Error : Parse Error: expected: '"' got: ':' . at ': {\n\
at parseEscapedItem: fast-csv\lib\parser\parser.js:80:19
at parseStream.parseLine: parse.js: 192:30
at parser_stream.js

We are using 4.1.0 version for Newman and this is being used as a library.
We are using Postman latest version 6.3.0 version. 

I would like to know is this issue with Newman and Postman versions or any bug from the library. This is similiar to the  request posted by @mianamirrashid.

Thanks in advance.

**Comment by shamasis** [comment]
@mianamirrashid @Manojkumar05 - are both of your issues related? It would be very difficult to resolve two issues on a single thread. If they aren’t related, I would urge @Manojkumar05 to file a separate issue please.

**Comment by jpavansharma** [comment]
@shamasis These two issues are related. Both of them have issues running collection in newman exported from Postman.
@codenirvana : I am too working with Manoj on running collection exported from Postman to Newman. Re verify from your side on Newman front. fast-csv library is used internally by Newman api. This is not done by us. From the command line too the collections exported from Newman are not running successfully. Please find the attached files with screen shots which shows 500 Internal server error.
[Newman runner issue.zip](https://github.com/postmanlabs/newman/files/2396423/Newman.runner.issue.zip)

**Comment by codenirvana** [comment]
@jpavansharma Thank you for the clarification. 

The collection supplied for testing was not portable since it had internal local endpoints. By modifying them to external endpoints, we were not able to reproduce the issue.

To summarise your problem, you are seeing script execution error and server 5xx when running collection. It seems to be an issue with the way things are scripted in the collection. I attempted to deep-dive the code written in the scripts and found a possible issue with your code:

In the second request's(**AddFolderTo Copy**) pre-request script:
on `line 5`, it's parsing the JSON data from the iterationData and the same is saved to an environment variable named `Input`, which is later used as request body for that request. The issue here is, you can't save JavaScript objects directly into an environment variable. You have to stringify it before saving, which in this case means you don't have to parse the already stringified value. If you check the request body sent for this request it will be `[object Object]` and it causes internal server error at your end.

I also observed that this collection does not work in Postman app as well. If it does, there could be differences between the two collections that we are missing here.

In addition to above, I am attaching [newman dependency tree](https://github.com/postmanlabs/newman/files/2399998/npm-tree.txt). Here, you can see that `fast-csv` library is not used by Newman. :-)

**Comment by shamasis** [comment]
Closing this issue for now, since this is not related to Newman. Postman also has extensive documentation that you can refer to quickly unblock yourselves from issues such as this. Visit: https://learning.getpostman.com/

Also, if your issue is more of an "how-to" help and not a product bug, we have an active user community forum that will help to solve them. Visit: https://community.getpostman.com/

In case your issue exists after making the suggested changes, please feel free to re-open the issue with a portable sample that works without needing to be changed.
